docs/advanced-features/custom-error-page.md

```diff
@@ -2,9 +2,34 @@
 description: Override and extend the built-in Error page to handle custom errors.
 ---
 
-# Custom Error Page
+## 404 Page
 
-**404** or **500** errors are handled both client-side and server-side by the `Error` component. If you wish to override it, define the file `pages/_error.js` and add the following code:
+A 404 page may be accessed very often. Rendering the `_error` page for every visit increases the load for your server or the invocations for serverless functions. This can result in increased costs and slow experiences.
+
+To avoid the above pitfalls, Next.js provides a static 404 page by default without having to add any additional files.
+
+The static 404 page is automatically statically optimized at build time and used with `next start` and when deploying on [ZEIT Now](https://zeit.co).
+
+If you have a custom `pages/_app.js` with `getInitialProps`, the automatic static optimization will not be able to be applied since you are defining a global data dependency.
+
+### Customizing The 404 Page
+
+To create a custom 404 page you can create a `pages/404.js` file. This file is meant to allow customizing the 404 page while keeping the page static and doesn't allow using any data-fetching methods like `getStaticProps`, `getServerSideProps`, or `getInitialProps`.
+
+```jsx
+// pages/404.js
+export default function custom404() {
+  return <h1>404 - Page Not Found</h1>
+}
+```
+
+## 500 Page
+
+By default Next.js provides an error page that matches the default 404 page's style. This page is not statically optimized like the 404 page and is why we allow creating a separate `pages/404.js` file to enable you to report 500 errors without de-optimizing your 404 page.
```

Comment:
I think this:

> is why we allow creating a separate `pages/404.js` file to enable you to report 500 errors without de-optimizing your 404 page.

Can be rephrased something like this:

```suggestion
By default Next.js provides a 500 error page that matches the default 404 page's style. Unlike the 404 page, the 500 page is not statically optimized because 500 errors happen dynamically. This is why we separate error handling for the 404 page and the 500 page.
```